contrib/platform/test/com/sun/jna/platform/win32/Crypt32Test.java

```diff
@@ -423,10 +423,22 @@ public void testCertVerifyCertificateChainPolicy() {
         assertTrue("The status would be true since a valid certificate chain was not passed in.", status);
     }
 
-    public void testCertEnumCertificatesInStore() {
-        String SYSTEM_STORE_NAME = "ROOT";
-        WinCrypt.HCERTSTORE hCertStore = Crypt32.INSTANCE.CertOpenSystemStore(null, SYSTEM_STORE_NAME);
+    public void testCertOpenSystemStore() {
+        WinCrypt.HCERTSTORE hCertStore = Crypt32.INSTANCE.CertOpenSystemStore(null, "ROOT");
+        enumerateRootCertificates(hCertStore);
+    }
+
+    public void testCertOpenStore() {
+        WinCrypt.HCERTSTORE hCertStore = Crypt32.INSTANCE.CertOpenStore(
+                new WTypes.LPSTR(new Pointer(Crypt32.CERT_STORE_PROV_SYSTEM_REGISTRY_W)),
```

Comment:
I'd say the entire function signature is quite ugly and in this PR the signature is official. That's what I prefer anyway instead of adding overloads

Reply:
"the signature is official" is a stretch in my mind. To me it looks silly to use an integer, build a pointer from that and pretend that its a pointer to char array. This is wrong on so many levels. This is not a criticism on the PR, but on the signature in itself. It would have been to easy if MS had just used names for the built-ins.
Depending on your decision I'll prototype the suggested alternative.

Reply:
Ok, I'm for better alternatives if we can cover all use cases

Reply:
I'm not sure if this is really better, but it is what I had in mind:

```java
    public static class CertStoreProviderName implements NativeMapped {
        private Pointer pointer;

        private CertStoreProviderName(Pointer fixedValue) {
            this.pointer = fixedValue;
        }

        public CertStoreProviderName(int fixedValue) {
            this.pointer = Pointer.createConstant(fixedValue);
        }

        public CertStoreProviderName(String name) {
            byte[] nameBytes = Native.toByteArray(name);
            this.pointer = new Memory(nameBytes.length);
            this.pointer.write(0, nameBytes, 0, nameBytes.length);
        }

        @Override
        public Object fromNative(Object nativeValue, FromNativeContext fnc) {
            if (nativeValue == null) {
                return null;
            }
            return new CertStoreProviderName((Pointer) nativeValue);
        }

        @Override
        public Object toNative() {
            return pointer;
        }

        @Override
        public Class<?> nativeType() {
            return Pointer.class;
        }

    }
```

This makes it clear, that the value should not be read. From the value alone you can't guess whether this is a hard coded value or a pointer to a char array at a very low memory location. A look at the implementation of this would be interesting.

I don't trust `LPSTR`. A quick look at `setValue` indicates, that this will fail hard on platforms where the system encoding is a not a single byte encoding (for example UTF-8).